pkg/virt-vmconfig/rest/endpoint.go

```diff
@@ -0,0 +1,76 @@
+package rest
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+	"k8s.io/apimachinery/pkg/api/errors"
+	k8sv1meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+)
+
+func StartFunc(request *restful.Request, response *restful.Response) {
+	name := request.PathParameter("name")
+	namespace := request.PathParameter("namespace")
+
+	virtCli, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	vmConfig, err := virtCli.VMConfig(namespace).Get(name, k8sv1meta.GetOptions{})
+	if errors.IsNotFound(err) {
+		response.WriteError(http.StatusNotFound, fmt.Errorf("VMConfig does not exist"))
+		return
+	}
+	if err != nil {
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	domainSpec := ConfigToSpec(vmConfig)
+	// TODO: Generate / supply different name, don't copy the VMConfig.
+	vm := v1.NewMinimalVM(vmConfig.ObjectMeta.Name)
+	vm.Spec.Domain = domainSpec
+
+	_, err = virtCli.VM(namespace).Create(vm)
```

Comment:
Actually the issue here is that @davidvossel talks about single VM per VMConfig and @mpolednik and the referenced 'config-and-instances.md' about multiple VMs per single VMConfig. It might make sense to first agree on what we are actually doing in high level. I tried to ask for the same in https://groups.google.com/forum/#!topic/kubevirt-dev/G0FpxJYFhf4 but the discussion has not really started..

Reply:
> Actually the issue here is that @davidvossel talks about single VM per VMConfig and @mpolednik and the referenced 'config-and-instances.md' about multiple VMs per single VMConfig. 

In config-and-instances.md there is this excerpt

```
A VirtualMachineConfig object is used to create one or more VM instances.
```

When I reviewed that document, my interpretation based on all the discussions that led up us collectively approving the document (and me merging the document) was that there would only be a single VM instance active for a VMConfig at a time. I explicitly called this out a few times. https://github.com/kubevirt/kubevirt/pull/272#pullrequestreview-48509441

The document's eventual reduction of scope was in response to this inconsistency (among other things)

Executing the "start" api call twice for a VMConfig that already has an active instance would result in a no-op.  The multiple instance part means that that single VMConfig may be responsible for starting and stopping multiple instances throughout the VMConfig object's lifecycle. Not that there are multiple instances active in parallel.

Example taken directly from that document.
```
$c1        $i1        $i2
POST
/start     POST
/stop      DELETE
PATCH
/start                POST
/stop                 DELETE
```
both i1 and i2 were created from config 1, but i1 and i2 did not exist at the same time. Starting the config when no active VM instance exists results in a new instance being created.

This is an important concept to agree upon because the API as defined by config-and-instances only accounts for a single VM per a VMConfig.  If we move into the territory of a VMConfig starting multiple instances in parallel, then we are really discussing a separate concept entirely.

Reply:
There is no reason for VMConfig to exist it's only used to spawn a single VM - in that case, everything could be embed into the VM object iteslf. If you believe the proposal is wrong in some sense, I don't understand why there's no PR open against it.

Reply:
the idea was to use VMConfig as a persistence mechanism for "transient" run-time only VM object.

Reply:
We can actually revisit if a persistent VM shoudl support multi or one VM.